### PR TITLE
test: add config recording set tests

### PR DESCRIPTION
**Pull request by joannalikespi**
Added additional tests
Refactored code to make it more testable:
- Extracted openFilePicker function outside of page to create a seam where a mock function can be passed in instead
- Added testIDs to buttons to allow them to be clicked during tests
Fixed bug:
- changed the display condition for the show-details button so it gets removed when a reclist is not selected

### CODE REVIEW

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx [hunk 1]
```diff
@@ -59,12 +112,130 @@ describe('ConfigureRecordingSetPage', () => {
...
+    );
+
     userEvent.selectOptions(screen.getByTestId('selectReclist'), 'デルタ式英語リストver5 (Delta English Ver. 5)');
     userEvent.click(await screen.findByTestId('show-details-button'));
 
     await screen.findByText('listContent');
 
     expect(recordingListDataService.readData).toHaveBeenCalledTimes(1);
+
+    userEvent.click(await screen.findByTestId('up-button'));
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: Why is this needed?

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx [hunk 2]
```diff
@@ -59,12 +112,130 @@ describe('ConfigureRecordingSetPage', () => {
...
+
+    userEvent.click(await screen.findByTestId('next-button'));
+    await screen.findByText('otoIni');
+
+    userEvent.click(await screen.findByTestId('next-button'));
+    await screen.findByText('voiceDvcfg');
+
+    expect(recordingListDataService.readData).toHaveBeenCalledTimes(1);
+
+    userEvent.click(await screen.findByTestId('up-button'));
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: Why is this needed?

File: src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import Row from 'react-bootstrap/Row';
 import { useTranslation } from 'react-i18next';
 
 import { Consumer } from '../../types';
-import { filename, openFilePicker } from '../../utils';
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx]: Why is this being passed in?

**Review by iluminar-yi — CHANGES_REQUESTED** [review]
Update PR title to follow commit message convention

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx (hunk 1 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: For some reason, the state wasn't resetting between tests, and I couldn't find a cleanup function that would reset it, so I had to manually go back to the main page between tests

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx (hunk 2 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: Same as previous

File: src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx (hunk 3 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx]: I refactored the code to extract the function out because I was unsure how to mock it when it was simply being used inside the component

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx [hunk 4]
```diff
@@ -59,12 +112,130 @@ describe('ConfigureRecordingSetPage', () => {
...
+    userEvent.selectOptions(screen.getByTestId('selectReclist'), '');
+
+    expect(screen.queryByTestId('show-details-button')).toBeFalsy();
+
+    // Click filepicker
+    userEvent.click(await screen.findByTestId('cust-list-button'));
+
+    expect(mockedFilePicker).toHaveBeenCalledTimes(1);
+
+    // Expect(recordingListDataService.readData).toHaveBeenCalledTimes(1);
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: Do you want to keep this `expect`?

File: src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx (hunk 3 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/set-recording-list-configuration.tsx]: Can you mock the `utils` module?

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx (hunk 1 above)
Comment (iluminar-yi) [reply src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: Have you tried `cleanup()` from react-testing-library?

**Review by iluminar-yi — CHANGES_REQUESTED** [review]
Please rebase with latest dev

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx (hunk 1 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: Yes, it wasn't actually resetting the state

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx [hunk 5]
```diff
@@ -1,17 +1,27 @@
...
 
 jest.mock('../../services/recording-list-data-service');
 // Mock the remote module to prevent import failure.
 jest.mock('@electron/remote', () => ({ dialog: jest.fn() }));
+jest.mock('../../utils', () => ({
+  ...jest.requireActual('../../utils'),
+  openFilePicker: jest.fn().mockImplementation(),
+}));
+
+afterEach(() => {
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: This should go inside the `describe` block, at the top.

**Review by iluminar-yi — CHANGES_REQUESTED** [review]

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx [hunk 6]
```diff
@@ -1,19 +1,29 @@
...
+import { openFilePicker } from '../../utils';
 
 import { ConfigureRecordingSetPage } from './configure-recording-set-page';
 
 jest.mock('../../services/recording-list-data-service');
 // Mock the remote module to prevent import failure.
 jest.mock('@electron/remote', () => ({ dialog: jest.fn() }));
+jest.mock('../../utils', () => ({
+  ...jest.requireActual('../../utils'),
+  openFilePicker: jest.fn().mockImplementation(),
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: If there are no implementations, then you don't need to call `.mockImplementation()`.

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx [hunk 7]
```diff
@@ -1,19 +1,29 @@
...
+}));
 
 describe('ConfigureRecordingSetPage', () => {
+  afterEach(() => {
+    cleanup();
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: Do you need to call `cleanup()`?

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx [hunk 8]
```diff
@@ -1,19 +1,29 @@
...
+    window.localStorage.clear();
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: Just `localStorage.clear();`

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx [hunk 9]
```diff
@@ -60,11 +144,38 @@ describe('ConfigureRecordingSetPage', () => {
...
+    expect.assertions(2);
+
+    const mockedFilePick = mocked(openFilePicker);
+    // Use of 'any' is required since method is overridden, and the default signature has a different return type than the one we're testing
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    mockedFilePick.mockResolvedValue('testReclist' as any);
+
+    render(<ConfigureRecordingSetPage onSettingsButtonClick={noop} onNext={noop} onBack={noop} onSetSelected={noop} />);
+    //Confirm no selected default reclist and no button
+
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: Remove extra line

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx [hunk 10]
```diff
@@ -60,11 +144,38 @@ describe('ConfigureRecordingSetPage', () => {
...
+  it('renders showDetails butten when custom reclist is selected', async () => {
+    expect.assertions(2);
+
+    const mockedFilePick = mocked(openFilePicker);
+    // Use of 'any' is required since method is overridden, and the default signature has a different return type than the one we're testing
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    mockedFilePick.mockResolvedValue('testReclist' as any);
+
+    render(<ConfigureRecordingSetPage onSettingsButtonClick={noop} onNext={noop} onBack={noop} onSetSelected={noop} />);
+    //Confirm no selected default reclist and no button
```
Comment (iluminar-yi) [inline src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx]: Space after `//`

**Review by iluminar-yi — CHANGES_REQUESTED** [review]

File: src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx (hunk 8 above)
Comment (joannalikespi) [reply src/renderer/components/configure-recording-set-page/configure-recording-set-page.test.tsx:24]: localStorage.clear() caused an error when I tried it initially

**Review by iluminar-yi — APPROVED** [review]
Please rebase
